### PR TITLE
feat(shrex/peer-manager): only add peers with compatible protocols

### PR DESCRIPTION
**Pull request by derrandz**
## Overview

This PR addresses #2604 

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [x] Linked issues closed with keywords

### CODE REVIEW

**Comment by derrandz** [comment]
I won't be able to add a label to the PR and make the CI happy given github perms.

File: share/p2p/peers/manager.go [hunk 1]
```diff
@@ -418,6 +428,31 @@ func (m *Manager) isBlacklistedPeer(peerID peer.ID) bool {
 	return !m.connGater.InterceptPeerDial(peerID)
 }
 
+func (m *Manager) arePeerProtocolsCompatible(peerID peer.ID) bool {
```
Comment (vgonkivs) [inline share/p2p/peers/manager.go:431]: I'd call it `isSupported()` or `supportsProtocols()`

File: share/p2p/peers/manager.go [hunk 2]
```diff
@@ -418,6 +428,31 @@ func (m *Manager) isBlacklistedPeer(peerID peer.ID) bool {
...
+	peerProtos, _ := m.host.Peerstore().GetProtocols(peerID)
```
Comment (vgonkivs) [inline share/p2p/peers/manager.go:432]: There is one more method in the peerStore that fits better in this case - `SupportsProtocols`

**Review by vgonkivs — CHANGES_REQUESTED** [review]
Hey Hamza, glad to see you. Left some comments

File: share/p2p/peers/manager.go (hunk 2 above)
Comment (derrandz) [reply share/p2p/peers/manager.go]: Thanks!

File: share/p2p/peers/manager.go (hunk 1 above)
Comment (derrandz) [reply share/p2p/peers/manager.go]: Cleaner, changed 👍🏻

**Comment by derrandz** [comment]
Thanks @vgonkivs 
Reiterated on this to include your suggestions, lmk! 👍🏻

**Review by renaynay — CHANGES_REQUESTED** [review]
@derrandz I left a comment on the associated issue - what we need is a way to retrieve peers from the peer manager `Get` for the specific protocol that's requesting the peer (e.g., shrex-eds needs a peer running the same shrex-eds protocol)

So `Get` needs to take a procolID and return a peer that supports it

Later on, we can optimise discovery for peers that "match best". Not in this PR though.

**Comment by vgonkivs** [comment]
N/A

**Pull request closed without merging**